File: packages/react-native/vision/android/src/main/java/expo/modules/vision/ExpoVisionModule.kt
package expo.modules.vision

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoVisionModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoVision")

    AsyncFunction("setValueAsync") { value: String ->
    }
  }
}

File: packages/react-native/vision/src/ExpoVisionModule.ts
import { NativeModule, requireNativeModule } from 'expo';

import type { Classification } from './ExpoVision.types';

declare class ExpoVisionModule extends NativeModule<{}> {
  classifyImage(uri: string, minConfidence: number): Promise<Classification[]>;
}

/** NOTE:
 * ExpoVisionModule.swift's ModuleDefinition Name
 * ExpoVisionModule.kt's ModuleDefinition Name
 */
export default requireNativeModule<ExpoVisionModule>('ExpoVision');
